File: Latihan Bareng rava/Kendaraan.java
abstract class Kendaraan {
    protected int nomor;
    protected int tahun;
    protected String merk;

    // Konstruktor default Kendaraan
    public Kendaraan() {
        this.nomor = 0;
        this.tahun = 0;
        this.merk = "XXX";
    }

    // Konstruktor Kendaraan dengan parameter
    public Kendaraan(int nomor, int tahun, String merk) {
        this.nomor = nomor;
        this.tahun = tahun;
        this.merk = merk;
    }

    // Method displayInfo
    public void printInfo() {
        System.out.println("Nomor Kendaraan: " + this.nomor);
        System.out.println("Tahun Keluaran: " + this.tahun);
        System.out.println("Merk: " + this.merk);
    }

    // Method biayaSewa
    abstract public int biayaSewa(int lamaSewa);
}

File: latihan 4 sekawan/Kendaraan.cpp
#include <iostream>
using namespace std;

#include "Kendaraan.hpp"

Kendaraan::Kendaraan(){
    this->merk="XXX";
    this->tahun=0;
    this->nomor=0;
    this->kategori="Mobil";
}

Kendaraan::Kendaraan(int nomor,int tahun, string merk, string kategori){
    this->merk=merk;
    this->tahun=tahun;
    this->nomor=nomor;
    this->kategori=kategori;
}

Kendaraan::Kendaraan(const Kendaraan&kendaraan){
    this->merk=kendaraan.merk;
    this->tahun=kendaraan.tahun;
    this->nomor=kendaraan.nomor;
    this->kategori=kendaraan.kategori;
}

Kendaraan::~Kendaraan(){}

void Kendaraan:: PrintInfo(){
    cout << "Nomor Kendaraan: " << this->nomor <<endl;
    cout << "Tahun Keluaran: " << this->tahun <<endl;
    cout << "Merk Kendaraan: " << this->merk <<endl;
    cout << "Kategori: " <<this->kategori <<endl;
 }

int Kendaraan::BiayaSewa(int lamaSewa) const{
    if (this->kategori=="Bus"){
        return 1000000*lamaSewa;
    }
    else if(this->kategori=="Mobil"){
        return 500000*lamaSewa;
    }
    else{
        if (lamaSewa<=5){
            return 5000000;
        }
        else{
            return 5000000+500000*(lamaSewa-5);
        }
    }
}

